internal/ethapi/api.go

```diff
@@ -2646,44 +2646,22 @@ func checkTxFee(gasPrice *big.Int, gas uint64, cap float64) error {
 
 // CheckTxFee exports a helper function used to check whether the fee is reasonable
 func CheckTxFee(ctx context.Context, backend CeloBackend, gasPrice *big.Int, gas uint64, feeCurrency *common.Address) error {
-	feeCap, err := ConvertTxFeeCapToCurrency(ctx, backend, feeCurrency)
-	if err != nil {
-		return err
-	}
-
-	return checkTxFee(gasPrice, gas, feeCap)
-}
-
-// ConvertTxFeeCapToCurrency converts FeeCap in native currency into the specified currency
-// If no fee currency is specified, it returns the native currency FeeCap as is
-func ConvertTxFeeCapToCurrency(ctx context.Context, backend CeloBackend, feeCurrency *common.Address) (float64, error) {
-	txFeeCap := backend.RPCTxFeeCap()
+	// This early return prevents unnecessary API calls and ensures tests that don't involve fee currency conversion pass
 	if feeCurrency == nil {
-		return txFeeCap, nil
+		return checkTxFee(gasPrice, gas, backend.RPCTxFeeCap())
 	}
 
 	rates, err := backend.GetExchangeRates(ctx, rpc.BlockNumberOrHashWithNumber(rpc.LatestBlockNumber))
 	if err != nil {
-		log.Warn("Failed to get exchange rates", "err", err)
-		return 0, err
-	}
-	rate, ok := rates[*feeCurrency]
-	if !ok {
-		return 0, fmt.Errorf("could not convert from native to fee currency (fee-currency=%s): %w ", feeCurrency, exchange.ErrUnregisteredFeeCurrency)
+		log.Warn("Failed to get exchange rates for latest block", "err", err)
```

Comment:
I believe this is overly verbose - especially since this warning can't be correlated to the request arguments and failed RPC requests are already logged with additional request-context:

https://github.com/celo-org/op-geth/blob/055f61b61a24fccd728b33e091d6ec4c13839d10/rpc/handler.go#L476-L481

So I'd just pass along the error, or wrap the error when the additional message should also be visible to the caller in the response.

Reply:
Removed logging because callee throws similar error 79ff008c97a4cba079adbbfdf4cedf3b0712e45e